### PR TITLE
[Core] update grpc to 1.46.6

### LINKED ISSUE #31956: [Dependency] Upgrade gRPC to 1.46.6
Upgrade to a version of gRPC that https://github.com/advisories/GHSA-cfmr-vrgj-vqwv in Zlib
* 1.46.6 has this patch: https://github.com/grpc/grpc/pull/31845

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

https://github.com/ray-project/ray/issues/31956

> Upgrade to a version of gRPC that https://github.com/advisories/GHSA-cfmr-vrgj-vqwv in Zlib
1.46.6 has this patch: https://github.com/grpc/grpc/pull/31845

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by cadedaniel** [comment]
wonder why they have [1.47.3](https://github.com/grpc/grpc/releases/tag/v1.47.3) released but not marked as latest...

**Review by cadedaniel — APPROVED** [review]

File: bazel/ray_deps_setup.bzl [hunk 1]
```diff
@@ -282,11 +282,11 @@ def ray_deps_setup():
         # https://github.com/grpc/grpc/blob/1ff1feaa83e071d87c07827b0a317ffac673794f/bazel/grpc_deps.bzl#L189
         # Ensure this rule matches the rule used by grpc's bazel/grpc_deps.bzl
         name = "boringssl",
-        sha256 = "e168777eb0fc14ea5a65749a2f53c095935a6ea65f38899a289808fb0c221dc4",
-        strip_prefix = "boringssl-4fb158925f7753d80fb858cb0239dff893ef9f15",
+        sha256 = "534fa658bd845fd974b50b10f444d392dfd0d93768c4a51b61263fd37d851c40",
+        strip_prefix = "boringssl-b9232f9e27e5668bc0414879dcdedb2a59ea75f2",
```
Comment (rkooo567) [inline bazel/ray_deps_setup.bzl:286]: Is this required to change for grpc upgrade?

**Review by rkooo567 — APPROVED** [review]

File: bazel/ray_deps_setup.bzl (hunk 1 above)
Comment (scv119) [reply bazel/ray_deps_setup.bzl:286]: yup according to the comments.

**Comment by ggydush** [comment]
Are there current issues with grpc>=1.50.0? Could the pin be relaxed?

**Pull request merged**